### PR TITLE
[NXP] Adding Zephyr examples

### PR DESCRIPTION
**Pull request by yeaissa**
Adding 3 examples:

*All-cluster-app
*Thermostat
*Laundry-washer-app

### CODE REVIEW

**Review by andy31415 — CHANGES_REQUESTED** [review]
Please add targets to build_examples.py (specifically add condifurations to `scripts/build/builders/nxp.py` and update `scripts/build/build/targets.py`.

We need everyone to be able to compile these examples.

**Comment by yeaissa** [comment]
> Please add targets to build_examples.py (specifically add condifurations to `scripts/build/builders/nxp.py` and update `scripts/build/build/targets.py`.
> 
> We need everyone to be able to compile these examples.

@andy31415 We made this PR https://github.com/project-chip/connectedhomeip/pull/34206 to add the docker image for those Zephyr examples. This is a preparation for what you requested.
`build_examples.py` + workflow support will be added to this PR soon.
Thanks

**Comment by yeaissa** [comment]
@andy31415 build for the 3 examples is now added to the build_examples.

File: scripts/build/build/targets.py [hunk 1]
```diff
@@ -499,22 +499,32 @@ def BuildNxpTarget():
...
         TargetPart('k32w0', board=NxpBoard.K32W0),
-        TargetPart('k32w1', board=NxpBoard.K32W1)
+        TargetPart('k32w1', board=NxpBoard.K32W1),
+        TargetPart('rw61x', board=NxpBoard.RW61X)
+    ])
+
+    # OS
+    target.AppendFixedTargets([
+        TargetPart('zephyr', os_env=NxpOsUsed.ZEPHYR).OnlyIfRe('rw61x'),
+        TargetPart('freertos', os_env=NxpOsUsed.FREERTOS).ExceptIfRe('rw61x'),
```
Comment (andy31415) [inline scripts/build/build/targets.py:509]: would this be clearer:

```python
target.AppendFixedTargets([
        TargetPart('k32w0-freertos', board=NxpBoard.K32W0, os_env=NxpOsUsed.FREERTOS),
        TargetPart('k32w1-freertos', board=NxpBoard.K32W1, os_env=NxpOsUsed.FREERTOS),
        TargetPart('rw61x-zephyr', board=NxpBoard.RW61X, os_env=NxpOsUsed.ZEPHYR),
    ])
```

this shows up nicer as `{k32w0-freertos,k32w1-freertos,rw61x-zephyr}` than `{k32w0,k32w1,rw61x}-{zephyr,freertos}` given that we have a forced either/or on the board.

File: scripts/build/build/targets.py [hunk 2]
```diff
@@ -499,22 +499,32 @@ def BuildNxpTarget():
...
     ])
 
     # apps
     target.AppendFixedTargets([
         TargetPart('lighting', app=NxpApp.LIGHTING).OnlyIfRe('(k32w0|k32w1)'),
-        TargetPart('contact-sensor', app=NxpApp.CONTACT).OnlyIfRe('(k32w0|k32w1)')
+        TargetPart('contact-sensor', app=NxpApp.CONTACT).OnlyIfRe('(k32w0|k32w1)'),
+        TargetPart('all-clusters', app=NxpApp.ALLCLUSTERS).OnlyIfRe('rw61x'),
+        TargetPart('laundry-washer', app=NxpApp.LAUNDRYWASHER).OnlyIfRe('rw61x'),
+        TargetPart('thermostat', app=NxpApp.THERMOSTAT).OnlyIfRe('rw61x')
```
Comment (andy31415) [inline scripts/build/build/targets.py:518]: should these regex logic be more zephyr vs freertos instead of on board?

**Review by andy31415 — APPROVED** [review]

File: scripts/build/build/targets.py (hunk 1 above)
Comment (yeaissa) [reply scripts/build/build/targets.py:509]: We plan to have the rw61x freertos support in this file in the future. So this platform can have a choice either to use freertos or zephyr, this is why I made it like this.

File: scripts/build/build/targets.py (hunk 2 above)
Comment (yeaissa) [reply scripts/build/build/targets.py:518]: I'm taking in consideration that rw61x freertos support can be added to this file later. rw61x freertos supports those 3 applications, so it makes sense to use the platform name in `OnlyIfRe`

File: scripts/build/build/targets.py (hunk 1 above)
Comment (andy31415) [reply scripts/build/build/targets.py:509]: Is there a concrete plan to have the `cross-support` (or "future" part) in the near/medium future?

I am looking if we should apply `You ain't gonna need it` here or not. Experience-wise I see about 80+% of things built "for the future" never happening.

Comment (axelnxp) [reply scripts/build/build/targets.py:509]: it is concrete, we are working on it internally, the goal is to merge all nxp platforms in a single builder.

**Review by axelnxp — APPROVED** [review]

**Pull request merged**